File: frontend/src/types/hardware.ts
export interface GPUCapabilities {
  hasIntelVA: boolean
  intelVA?: VAInfo
  hasNVIDIA: boolean
  hasAMD: boolean
}

export interface VAInfo {
  profileCount: number
  entrypointCount: number
  decodeProfiles: string[]
  encodeProfiles: string[]
}

File: backend/internal/model/hardware.go
package model

// GPUCapabilities represents GPU encoding and decoding capabilities
type GPUCapabilities struct {
	HasIntelVA bool     `json:"hasIntelVA"`
	IntelVA    *VAInfo  `json:"intelVA,omitempty"`
	HasNVIDIA  bool     `json:"hasNVIDIA"`
	HasAMD     bool     `json:"hasAMD"`
}

// VAInfo represents Intel VA-API capabilities
type VAInfo struct {
	ProfileCount    int      `json:"profileCount"`
	EntrypointCount int      `json:"entrypointCount"`
	DecodeProfiles  []string `json:"decodeProfiles"`
	EncodeProfiles  []string `json:"encodeProfiles"`
}

